### PR TITLE
Adds put/post with data store.

### PR DESCRIPTION
**Pull request by saralein**

### CODE REVIEW

File: src/main/java/com/saralein/server/connection/ConnectionSocket.java [hunk 1]
```diff
@@ -21,9 +21,20 @@ public String read() throws IOException {
...
+
+            if (str.startsWith("Content-Length")) {
+                length = Integer.parseInt(str.split(":")[1].trim());
+            }
+        }
+
+        if (length > 0) {
+            char[] body = new char[length];
+            bufferedReader.read(body);
+            request.append(CRLF + "Body: " + new String(body));
```
Comment (saralein) [inline src/main/java/com/saralein/server/connection/ConnectionSocket.java]: This is the spot where I'm trying to decide if this should move to the parser, which would then receive a `BufferedReader` instead.

File: src/test/java/com/saralein/server/controller/form/FormControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private FormStore formStore;
+
+    @Before
+    public void setUp() {
+        String body = "<h3>My: Data</h3><h3>More: Stuff</h3>";
+
+        bodyArray = body.getBytes();
+
+        RequestParser requestParser = new RequestParser();
+        Request request = requestParser.parse("POST /form HTTP/1.1\r\n\r\nBody: My=Data&More=Stuff");
```
Comment (zdavy) [inline src/test/java/com/saralein/server/controller/form/FormControllerTest.java]: Feels like a lot of tests are tightly coupled to your requestParser implementation, is there a more expressive way to build a request?

Comment (saralein) [reply src/test/java/com/saralein/server/controller/form/FormControllerTest.java]: Maybe I will look into another builder between the parser and request.  They have made other areas easier to work with. <- Just going to give it a hash.

Comment (saralein) [reply src/test/java/com/saralein/server/controller/form/FormControllerTest.java]: Alright, made some updates to this on the [latest commit](https://github.com/saralein/java-server/pull/6/commits/bdbd93e446780008b4c8fd5a9f2e3699618a3238).

**Review by zdavy — APPROVED** [review]

File: src/test/java/com/saralein/server/request/RequestParserTest.java [hunk 3]
```diff
@@ -6,40 +6,55 @@
...
+        fileRequest = new Request(fileHash);
+        postRequest = new Request(postHash);
 
         requestParser = new RequestParser();
     }
 
     @Test
     public void returnsRequestWithCorrectInformation() {
-        Request test1ThroughParser = requestParser.parse("GET / HTTP/1.1");
+        Request parsedRootRequest = requestParser.parse("GET / HTTP/1.1");
```
Comment (damonkelley) [inline src/test/java/com/saralein/server/request/RequestParserTest.java]: It feels like you might have a couple of tests here

**Review by damonkelley — APPROVED** [review]

File: src/test/java/com/saralein/server/request/RequestParserTest.java (hunk 3 above)
Comment (saralein) [reply src/test/java/com/saralein/server/request/RequestParserTest.java]: Missed this.  I'll fix it and then merge.  Thank you.

**Pull request merged**